### PR TITLE
Run System.Diagnostics.StackTrace tests with NAOT

### PR DESCRIPTION
**Pull request by MichalStrehovsky**
Also fixes some corner case issues, like whether an empty stacktrace should stringify into a newline.

Cc @dotnet/ilc-contrib

### CODE REVIEW

File: src/libraries/System.Diagnostics.StackTrace/tests/StackFrameTests.cs [hunk 1]
```diff
@@ -123,6 +123,7 @@ public static IEnumerable<object[]> ToString_TestData()
 
         [Theory]
         [ActiveIssue("https://github.com/mono/mono/issues/15186", TestRuntimes.Mono)]
+        [ActiveIssue("TODOTODOTODO", typeof(PlatformDetection), nameof(PlatformDetection.IsNativeAot))]
```
Comment (MichalStrehovsky) [inline src/libraries/System.Diagnostics.StackTrace/tests/StackFrameTests.cs:126]: Now I remember why I didn't commit this last night. No merge before this is replaced.

File: src/coreclr/nativeaot/System.Private.CoreLib/src/System/Diagnostics/StackFrame.NativeAot.cs [hunk 2]
```diff
@@ -110,14 +110,26 @@ private void InitializeForIpAddress(IntPtr ipAddress, bool needFileInfo)
...
+
             const int SystemDiagnosticsStackDepth = 2;
 
-            frameIndex += SystemDiagnosticsStackDepth;
-            IntPtr[] frameArray = new IntPtr[frameIndex + 1];
-            int returnedFrameCount = RuntimeImports.RhGetCurrentThreadStackTrace(frameArray);
-            int realFrameCount = (returnedFrameCount >= 0 ? returnedFrameCount : frameArray.Length);
+            // Unreasonably high or negative frameIndex could lead to overflows or failures to allocate
+            // the frameArray below so spot check it's sane.
+            RuntimeImports.RhGetCurrentThreadStackBounds(out IntPtr stackLow, out IntPtr stackHigh);
```
Comment (jkotas) [inline src/coreclr/nativeaot/System.Private.CoreLib/src/System/Diagnostics/StackFrame.NativeAot.cs:119]: Instead of this, would it be better for `RhGetCurrentThreadStackTrace` to take an integer with how many frames to skip?

Comment (MichalStrehovsky) [reply src/coreclr/nativeaot/System.Private.CoreLib/src/System/Diagnostics/StackFrame.NativeAot.cs]: I reverted this part and block the relevant tests on #103218. This API is really confusing wrt integer overflows and underflows (that we enshrine in tests rather than block) and propagating the confusion to the runtime APIs doesn't sound like an improvement. It should be dealt separately from this PR.

**Review by jkotas — APPROVED** [review]
